unstructured/partition/text_type.py

```diff
@@ -291,7 +294,17 @@ def exceeds_cap_ratio(text: str, threshold: float = 0.5) -> bool:
         return True
 
     capitalized = sum([word.istitle() or word.isupper() for word in tokens])
-    ratio = capitalized / len(tokens)
+    # NOTE(klaijan) - detect the first word in the sentence, whether it is alpha or not
+    is_first_word_capitalized = (text.split()[0]).istitle()
+    ne_chunks = ne_chunk(text)
+    count_ne = sum([hasattr(chunk, "label") * len(chunk) for chunk in ne_chunks])
+
+    if len(tokens) > is_first_word_capitalized + count_ne:
+        ratio = (capitalized - is_first_word_capitalized - count_ne) / (
+            len(tokens) - is_first_word_capitalized - count_ne
+        )
```

Comment:
- `is_first_word_capitalized + count_ne` used multiple times so might as well compute once and store in one variable (also can consolidate lines 298-300)
- can we have a test for this case please? The existing tests checks if first word is capitalized but doesn't check the entity capitalized case. 
- careful with edge case where every token has `"label"` -> division by 0

Reply:
> * `is_first_word_capitalized + count_ne` used multiple times so might as well compute once and store in one variable (also can consolidate lines 298-300)
I thought having them separate is easier to follow the logic or make and adjustments moving forward as they are two different cases. I can consolidate line 299-300 together and combine `ne_chunk` variables into the `count_ne`

> * can we have a test for this case please? The existing tests checks if first word is capitalized but doesn't check the entity capitalized case.
Will do!

> * careful with edge case where every token has `"label"` -> division by 0
The edge case was captured with if-else. If the overall token lengths exceed the subtracted part, then we apply the new calculation, else it uses the same calculation. Allow me to think through this a little bit more whether the original calculation is always the best case to fall to or not.

Reply:
Also I noticed that I might have missed the case where NE is also the first word. I'll add on that.